galsim/wfirst/wfirst_backgrounds.py

```diff
@@ -0,0 +1,140 @@
+#
+"""
+@file wfirst_backgrounds.py
+
+Part of the WFIRST module.  This file includes any routines needed to define the background level,
+for which the main contribution is zodiacal light.
+"""
+
+import galsim
+import numpy as np
+import os
+
+def getSkyLevel(bandpass, world_pos=None, exptime=None, epoch=2015, julian_date=None):
+    """
+    Get the expected sky level for a WFIRST observation due to zodiacal light for this bandpass and
+    position.
+
+    This routine requires Bandpass objects that were loaded by galsim.wfirst.getBandpasses().  That
+    routine will have stored tables containing the sky background as a function of position on the
+    sky for that bandpass.  This routine then interpolates between the values in those tables to
+    arbitrary positions on the sky.
+
+    The numbers that are stored in the Bandpass object `bandpass` are background level in units of
+    e-/m^2/s/arcsec^2.  To get rid of the m^2, this routine multiplies by the total effective
+    collecting area in m^2.  Multiplying by the exposure time gives a result in e-/arcsec^2.  The
+    result can either be multiplied by the approximate pixel area to get e-/pix, or the result can
+    be used with wcs.makeSkyImage() to make an image of the sky that properly includes the actual
+    pixel area as a function of position on the detector.
+
+    The source of the tables that are being interpolated is Chris Hirata's publicly-available WFIRST
+    exposure time calculator (ETC):
+
+        http://www.tapir.caltech.edu/~chirata/web/software/space-etc/
+
+    It nominally returns photons/m^2/s/arcsec^2, but the input bandpasses used internally by the ETC
+    code include the quantum efficiency, to effectively convert to e-/m^2/s/arcsec^2.  Note that in
+    general results will depend on the adopted model for zodiacal light, and these are uncertain at
+    the ~10% level.
+
+    Positions should be specified with the `world_pos` keyword, which must be a CelestialCoord
+    object.  If no `world_pos` is supplied, then the routine will use a default position that looks
+    sensibly away from the sun.
+
+    @param bandpass     A Bandpass object.
+    @param world_pos    A position, given as a CelestialCoord object.  If None, then the routine
+                        will use an ecliptic longitude of 90 degrees with respect to the sun
+                        position (as a fair compromise between 0 and 180), and an ecliptic latitude
+                        of 30 degrees with respect to the sun position (decently out of the plane
+                        of the Earth-sun orbit). [default: None]
+    @param exptime      Exposure time in seconds.  If None, use the default WFIRST exposure time.
+                        [default: None]
+    @param epoch        The epoch to be used for estimating the obliquity of the ecliptic when
+                        converting to ecliptic coordinates.
+                        [default: 2015.]
```

Comment:
Also, maybe it's better to combine epoch and julian_date here.  (Not in celestial.py!)  The the user passes in a real python date or datetime object, then we can get both the epoch and the julian_date from that.  Then the default could be some particular date like `datetime.date(2020,3,23)`.